components/contribution-flow/index.js

```diff
@@ -277,34 +277,14 @@ class CreateOrderPage extends React.Component {
     }
 
     // sets the total amount & fee options correctly when we change the base amount (if fees on top is available)
-    if (
-      prevState.stepDetails?.amount !== this.state.stepDetails?.amount &&
-      this.props.feesOnTopAvailable &&
-      this.state.stepProfile?.type !== 'COLLECTIVE' &&
-      this.props.tier?.type !== 'TICKET'
-    ) {
-      this.setState({ platformFeeOptions: this.createplatformFeeOptions(this.state.stepDetails.amount) });
+    const hasProfileChanged = prevState.stepProfile?.id !== this.state.stepProfile?.id;
+    const hasAmountChanged = prevState.stepDetails?.amount !== this.state.stepDetails?.amount;
+    if ((hasProfileChanged || hasAmountChanged) && this.canHaveFeesOnTop()) {
+      const platformFeeOptions = this.createplatformFeeOptions(this.state.stepDetails.amount);
       this.setState(state => ({
-        stepDetails: {
-          ...state.stepDetails,
-          platformFee: this.createplatformFeeOptions(this.state.stepDetails.amount)[1],
-        },
+        platformFeeOptions: platformFeeOptions,
+        stepDetails: { ...state.stepDetails, platformFee: platformFeeOptions[1] },
       }));
-      // sets the total amount correctly the first time we load step 2 (if fees on top is available)
-      if (
-        prevState.stepDetails?.platformFee !== this.state.stepDetails?.platformFee &&
-        this.props.feesOnTopAvailable &&
-        this.state.stepProfile?.type !== 'COLLECTIVE' &&
-        this.props.tier?.type !== 'TICKET'
-      ) {
-        this.setState(state => ({
-          stepDetails: {
-            ...state.stepDetails,
-            totalAmount:
-              state.stepDetails.totalAmount + this.createplatformFeeOptions(this.state.stepDetails.amount)[1],
-          },
-        }));
-      }
```

Comment:
My understanding was that this code was unreachable, because it's nested in the other `if` that checks if amount has changed. So we'd have to change amount and platformFee at the same time to reach it, which I don't think is possible. @sbinlondon let me know if I missed something